### PR TITLE
Add metadata to compute plan

### PR DESCRIPTION
**Pull request by maeldebon**
:link: **Related Pull Requests :**
• [substra](https://github.com/SubstraFoundation/substra/pull/189)
• [substra-tests](https://github.com/SubstraFoundation/substra-tests/pull/112)
• [substra-chaincode](https://github.com/SubstraFoundation/substra-chaincode/pull/121)

This PR aims to add a metadata field to the compute plan asset.

### CODE REVIEW

File: backend/substrapp/serializers/ledger/computeplan/serializer.py [hunk 1]
```diff
@@ -75,6 +81,8 @@ def get_args(self, data):
             }
             if 'in_models_ids' in data_traintuple:
                 traintuple['inModelsIDs'] = data_traintuple['in_models_ids']
+            if 'metadata' in data_traintuple:
+                traintuple['metadata'] = data_traintuple['metadata']
```
Comment (jmorel) [inline backend/substrapp/serializers/ledger/computeplan/serializer.py:85]: Is there a reason to only include metadata if present vs always including it like we do for the other assets (datasets, objectives etc.) ?

File: backend/substrapp/tests/views/tests_views_algo.py [hunk 2]
```diff
@@ -218,6 +222,7 @@ def test_algo_create(self):
             mcreate.return_value = {}
             response = self.client.post(url, data=data, format='multipart', **self.extra)
 
+        assert response.data['metadata'] is not None
```
Comment (jmorel) [inline backend/substrapp/tests/views/tests_views_algo.py:225]: Why not test that it is equal to the actual value ?

Comment (maeldebon) [reply backend/substrapp/tests/views/tests_views_algo.py:225]: I just checked, we cannot get the metadata's value by using the response object. I'll remove this test for now.

File: backend/substrapp/serializers/ledger/computeplan/serializer.py (hunk 1 above)
Comment (maeldebon) [reply backend/substrapp/serializers/ledger/computeplan/serializer.py:85]: Well, it doesn't work when I change this, so I suppose it's the right way to do it.

**Pull request merged**